### PR TITLE
docs: add EventManager examples

### LINKED ISSUE #12: chore: Implement user scene post-initialization event
It should be an event fired when the game.ts code has run and the scene is ready to render stuff.

### PR DESCRIPTION
**Pull request by menduz**
Add a EventManager example.
Adds scene created event, closes #12

### CODE REVIEW

File: packages/decentraland-ecs/src/decentraland/Implementation.ts [hunk 1]
```diff
@@ -42,10 +42,8 @@ export class DecentralandSynchronizationSystem implements ISystem {
     this.dcl.onEvent(event => {
       switch (event.type) {
         case 'uuidEvent':
-          const e = new UUIDEvent()
-          e.uuid = (event.data as any).uuid
-          e.payload = (event.data as any).payload
-          engine.eventManager.fireEvent(e)
+          const data = event.data as any
```
Comment (fmiras) [inline packages/decentraland-ecs/src/decentraland/Implementation.ts:45]: Wouldn't be cooler to make
```ts
const { data }: any = event
```
so we cast `event.data` object to `any` type once?

File: test/unit/ecs.test.tsx [hunk 2]
```diff
@@ -1071,6 +1062,18 @@ describe('ECS', () => {
...
+      const materials: PBRMaterial[] = []
+
+      for (let [, x] of context.disposableComponents) {
+        if (x instanceof PBRMaterial) {
+          materials.push(x)
+        }
+      }
+
+      return materials
+    }
```
Comment (fmiras) [inline test/unit/ecs.test.tsx:1075]: Just a comment, this function could be lot more readable if separate them in lambda functions like this:

```ts
function getMaterials(context: SharedSceneContext): DisposableComponent[] {
	return Array.from(context.disposableComponents)
    	.map(pair => pair[1]) // or .map(([, component]) => component)
        .filter(component => component instanceof PBRMaterial)
}
```

or even just asking for the entries:
```ts
function getMaterials(context: SharedSceneContext): DisposableComponent[] {
	return Array.from(context.disposableComponents.entries())
        .filter(component => component instanceof PBRMaterial)
}

```

File: packages/decentraland-ecs/src/ecs/EventManager.ts [hunk 3]
```diff
@@ -96,19 +101,18 @@ export class EventManager {
...
   if (takenEventNames.indexOf(eventName) !== -1) {
     throw new Error(`The event name ${eventName} is already taken`)
   }
 
   takenEventNames.push(eventName)
 
-  return (target: any) => {
-    target[eventNameSymbol] = eventName
+  return <TFunction extends Function>(target: TFunction): TFunction | void => {
+    // tslint:disable-next-line:semicolon
```
Comment (fmiras) [inline packages/decentraland-ecs/src/ecs/EventManager.ts:114]: Not related to the PR but what do you think of removing this rule? prettier already remove unnecessary semi-colons for us and we already have 16 matching results of this comment. Is not bad to use it

**Review by fmiras — CHANGES_REQUESTED** [review]
LGTM but I'd to discuss the comments I left

File: packages/decentraland-ecs/src/decentraland/Implementation.ts (hunk 1 above)
Comment (menduz) [reply packages/decentraland-ecs/src/decentraland/Implementation.ts:45]: That does not produce the same result

File: packages/decentraland-ecs/src/ecs/EventManager.ts (hunk 3 above)
Comment (menduz) [reply packages/decentraland-ecs/src/ecs/EventManager.ts:114]: Sure, we should wait to merge the open things before, the merge conflicts will kill us

File: test/unit/ecs.test.tsx (hunk 2 above)
Comment (menduz) [reply test/unit/ecs.test.tsx:1075]: Is it that bad? I like functional thing when they save us dozens of lines, without sacrificing readability. This one is pretty clear

**Comment by nearnshaw** [comment]
@menduz  

I'm documenting events right now.
Will we keep the second argument of the `addListener` function?  I think I remember a conversation we had where you said that was pointless and you'd try to remove it.

```
events.addListener(UpdateEvent, null, ({ entity, dt }) => {
    // function
})
```

Will it remain there for the release?

**Comment by menduz** [comment]
Yes, it will remain for performance reasons.

**Pull request merged**